stock-tracker/src/features/chart/redux/chartMiddleware.js

```diff
@@ -0,0 +1,18 @@
+import { SET_CHART_TIME } from "./actionTypes";
+import { CHART_TIME } from "../../../socket/eventTypes";
+
+export const chartMiddleware = ({
+  socketService
+}) => store => next => action => {
+  const result = next(action);
```

Comment:
Why do you call `next(action)` at the start, and not at the end like `return next(action)`

Same with all your other middleware.

Reply:
Have had a look and changed it so I am now using the payload to emit rather than getting from store

Reply:
Eh? I don't get it...